app/models/ConsignmentLevelDocuments.scala

```diff
@@ -16,23 +16,25 @@
 
 package models
 
-import config.FrontendAppConfig
+import config.{FrontendAppConfig, PhaseConfig}
 import pages.document._
 import pages.sections.DocumentsSection
+import play.api.Configuration
 
 case class ConsignmentLevelDocuments(
   previous: Int,
   supporting: Int,
   transport: Int
 ) {
 
-  def canAdd(documentType: DocumentType)(implicit config: FrontendAppConfig): Boolean = documentType match {
-    case DocumentType.Previous  => previous < config.maxPreviousDocuments
-    case DocumentType.Support   => supporting < config.maxSupportingDocuments
-    case DocumentType.Transport => transport < config.maxTransportDocuments
-  }
+  def canAdd(documentType: DocumentType)(implicit config: FrontendAppConfig, phaseConfig: PhaseConfig, configuration: Configuration): Boolean =
+    documentType match {
+      case DocumentType.Previous  => previous < phaseConfig.maxPreviousDocuments
```

Comment:
If we keep this as `config.maxPreviousDocuments` then we won't need to inject `Configuration` everywhere. In `FrontendAppConfig` we can have
```
def maxPreviousDocuments(implicit phaseConfig: PhaseConfig) = phaseConfig.maxPreviousDocuments(configuration)

```